lib/api/utils.rb

```diff
@@ -5,5 +5,15 @@ def self.build_href_slug(klass, id)
       collection = Api::CollectionConfig.new.name_for_subclass(klass)
       "#{collection}/#{id}" if collection
     end
+
+    def self.resource_search_by_href_slug(href_slug)
```

Comment:
@abellotti can we pass in the user_object here with the default being User.curent_user

```
def self.resource_search_by_href(href_slug, user = User.current_user)
```